### ISSUE TITLE
Error in counterfactual for NLP model

### ISSUE THREAD

**Issue by AmirMK** [salesforce/OmniXAI#61]
I need to generate counterfactual results for my sentiment analysis model and I tried to use OmniXIA by replicating this example: https://opensource.salesforce.com/OmniXAI/latest/tutorials/nlp/ce_classification.html however I am getting below error while generating the counterfactual. I am using python 3.9 and the latest version of OmniXIA (1.2.3) 

```
INFO:polyjuice.polyjuice_wrapper:Setup Polyjuice.
INFO:polyjuice.polyjuice_wrapper:Setup SpaCy processor.
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[5], line 9
      1 x = Text([
      2     "What a great movie! if you have no taste.",
      3     "it was a fantastic performance!",
   (...)
      7     "i've never watched something as bad"
      8 ])
----> 9 explanations = explainer.explain(x)

File ~\Anaconda3\envs\omnixai\lib\site-packages\omnixai\explainers\nlp\counterfactual\polyjuice.py:173, in Polyjuice.explain(self, X, max_number_examples, **kwargs)
    160 """
    161 Generates the counterfactual explanations for the input instances.
    162 
   (...)
    170 :return: The explanations for all the input instances.
    171 """
    172 if self.mode == "classification":
--> 173     return self._explain_classification(X=X, max_number_examples=max_number_examples, **kwargs)
    174 else:
    175     return self._explain_question_answering(X=X, max_number_examples=max_number_examples, **kwargs)

File ~\Anaconda3\envs\omnixai\lib\site-packages\omnixai\explainers\nlp\counterfactual\polyjuice.py:96, in Polyjuice._explain_classification(self, X, max_number_examples, **kwargs)
     94 for idx, text in enumerate(X.values):
     95     original_label = labels[idx]
---> 96     perturb_texts = self._perturb(text.lower(), **kwargs)
     97     perturb_texts = list(set([t.lower() for t in perturb_texts]))
     98     perturb_predictions, perturb_labels = self._predict(
     99         Text(perturb_texts, tokenizer=tokenizer))

File ~\Anaconda3\envs\omnixai\lib\site-packages\omnixai\explainers\nlp\counterfactual\polyjuice.py:73, in Polyjuice._perturb(self, text, **kwargs)
     71 ce_type = kwargs.get("ce_type", "perturb")
     72 if ce_type == "perturb":
---> 73     perturb_texts = self.explainer.perturb(
     74         text,
     75         num_perturbations=kwargs.get("num_perturbations", 10),
     76         perplex_thred=kwargs.get("perplex_thred", 10)
     77     )
     78 elif ce_type == "blank":
     79     perturb_texts = self.explainer.get_random_blanked_sentences(
     80         sentence=text,
     81         max_blank_sent_count=kwargs.get("num_perturbations", 10),
     82         is_token_only=True,
     83         max_blank_block=1
     84     )

File ~\Anaconda3\envs\omnixai\lib\site-packages\polyjuice\polyjuice_wrapper.py:247, in Polyjuice.perturb(self, orig_sent, blanked_sent, is_complete_blank, ctrl_code, perplex_thred, num_perturbations, verbose, **kwargs)
    245     logger.info("Generating on these prompts:")
    246     for p in prompts: logger.info(f" | {p}")
--> 247 generated = generate_on_prompts(
    248     generator=self.generator, prompts=prompts, **kwargs)
    249 merged = list(np.concatenate(generated))
    251 validated_set = []

File ~\Anaconda3\envs\omnixai\lib\site-packages\polyjuice\generations\generator_helpers.py:59, in generate_on_prompts(generator, prompts, temperature, num_beams, n, do_sample, batch_size, **kwargs)
     57 def generate_on_prompts(generator, prompts, temperature=1, 
     58     num_beams=None, n=3, do_sample=True, batch_size=128, **kwargs):
---> 59     preds_list = batched_generate(generator, prompts,
     60         temperature=temperature, n=n, 
     61         num_beams=num_beams, 
     62         do_sample=do_sample, batch_size=batch_size, **kwargs)
     63     if len(prompts) == 1:
     64         preds_list = [preds_list]

File ~\Anaconda3\envs\omnixai\lib\site-packages\polyjuice\generations\generator_helpers.py:44, in batched_generate(generator, examples, temperature, num_beams, num_return_sequences, do_sample, batch_size, **kwargs)
     42 with torch.no_grad():
     43     for e in (range(0, len(examples), batch_size)):
---> 44         preds_list += generator(
     45             examples[e:e+batch_size],
     46             temperature=temperature,
     47             return_tensors=True,
     48             num_beams=num_beams,
     49             max_length=1000,
     50             early_stopping=None if num_beams is None else True,
     51             do_sample=num_beams is None and do_sample,
     52             num_return_sequences=num_return_sequences, 
     53             **kwargs
     54         )
     55 return preds_list

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\pipelines\text_generation.py:202, in TextGenerationPipeline.__call__(self, text_inputs, **kwargs)
    163 def __call__(self, text_inputs, **kwargs):
    164     """
    165     Complete the prompt(s) given as inputs.
    166 
   (...)
    200           ids of the generated text.
    201     """
--> 202     return super().__call__(text_inputs, **kwargs)

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\pipelines\base.py:1063, in Pipeline.__call__(self, inputs, num_workers, batch_size, *args, **kwargs)
   1059 if can_use_iterator:
   1060     final_iterator = self.get_iterator(
   1061         inputs, num_workers, batch_size, preprocess_params, forward_params, postprocess_params
   1062     )
-> 1063     outputs = [output for output in final_iterator]
   1064     return outputs
   1065 else:

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\pipelines\base.py:1063, in <listcomp>(.0)
   1059 if can_use_iterator:
   1060     final_iterator = self.get_iterator(
   1061         inputs, num_workers, batch_size, preprocess_params, forward_params, postprocess_params
   1062     )
-> 1063     outputs = [output for output in final_iterator]
   1064     return outputs
   1065 else:

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\pipelines\pt_utils.py:124, in PipelineIterator.__next__(self)
    121     return self.loader_batch_item()
    123 # We're out of items within a batch
--> 124 item = next(self.iterator)
    125 processed = self.infer(item, **self.params)
    126 # We now have a batch of "inferred things".

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\pipelines\pt_utils.py:125, in PipelineIterator.__next__(self)
    123 # We're out of items within a batch
    124 item = next(self.iterator)
--> 125 processed = self.infer(item, **self.params)
    126 # We now have a batch of "inferred things".
    127 if self.loader_batch_size is not None:
    128     # Try to infer the size of the batch

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\pipelines\base.py:990, in Pipeline.forward(self, model_inputs, **forward_params)
    988     with inference_context():
    989         model_inputs = self._ensure_tensor_on_device(model_inputs, device=self.device)
--> 990         model_outputs = self._forward(model_inputs, **forward_params)
    991         model_outputs = self._ensure_tensor_on_device(model_outputs, device=torch.device("cpu"))
    992 else:

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\pipelines\text_generation.py:244, in TextGenerationPipeline._forward(self, model_inputs, **generate_kwargs)
    242 prompt_text = model_inputs.pop("prompt_text")
    243 # BS x SL
--> 244 generated_sequence = self.model.generate(input_ids=input_ids, attention_mask=attention_mask, **generate_kwargs)
    245 out_b = generated_sequence.shape[0]
    246 if self.framework == "pt":

File ~\Anaconda3\envs\omnixai\lib\site-packages\torch\autograd\grad_mode.py:27, in _DecoratorContextManager.__call__.<locals>.decorate_context(*args, **kwargs)
     24 @functools.wraps(func)
     25 def decorate_context(*args, **kwargs):
     26     with self.clone():
---> 27         return func(*args, **kwargs)

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\generation\utils.py:1296, in GenerationMixin.generate(self, inputs, max_length, min_length, do_sample, early_stopping, num_beams, temperature, penalty_alpha, top_k, top_p, typical_p, repetition_penalty, bad_words_ids, force_words_ids, bos_token_id, pad_token_id, eos_token_id, length_penalty, no_repeat_ngram_size, encoder_no_repeat_ngram_size, num_return_sequences, max_time, max_new_tokens, decoder_start_token_id, use_cache, num_beam_groups, diversity_penalty, prefix_allowed_tokens_fn, logits_processor, renormalize_logits, stopping_criteria, constraints, output_attentions, output_hidden_states, output_scores, return_dict_in_generate, forced_bos_token_id, forced_eos_token_id, remove_invalid_values, synced_gpus, exponential_decay_length_penalty, suppress_tokens, begin_suppress_tokens, forced_decoder_ids, **model_kwargs)
   1294 # 0. Validate the `.generate()` call
   1295 self._validate_model_class()
-> 1296 self._validate_model_kwargs(model_kwargs.copy())
   1298 # 1. Set generation parameters if not already defined
   1299 bos_token_id = bos_token_id if bos_token_id is not None else self.config.bos_token_id

File ~\Anaconda3\envs\omnixai\lib\site-packages\transformers\generation\utils.py:993, in GenerationMixin._validate_model_kwargs(self, model_kwargs)
    990         unused_model_args.append(key)
    992 if unused_model_args:
--> 993     raise ValueError(
    994         f"The following `model_kwargs` are not used by the model: {unused_model_args} (note: typos in the"
    995         " generate arguments will also show up in this list)"
    996     )

ValueError: The following `model_kwargs` are not used by the model: ['n'] (note: typos in the generate arguments will also show up in this list)
```

**Comment by yangwenz** [comment]
Which version of polyjuice has been used?

**Comment by AmirMK** [comment]
polyjuice version is 0.1.5. I got it installed via `pip install omnixai[nlp]`

**Comment by yangwenz** [comment]
The issue has been fixed, please update to v1.2.4.
